pkg/apis/core/v1alpha1/cluster_types.go

```diff
@@ -79,12 +79,58 @@ type KubernetesClusterConnection struct {
 	// The name of the kubeconfig context to use.
 	//
 	// If not specified, will use the default context in the kubeconfig.
+	//
+	// +optional
 	Context string `json:"context,omitempty" protobuf:"bytes,1,opt,name=context"`
 
 	// The default namespace to use.
 	//
 	// If not specified, will use the namespace in the kubeconfig.
+	//
+	// +optional
 	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
+
+	// DefaultRegistryOptions determines where images for this Cluster should
```

Comment:
it might make more sense to put this on the Cluster.Spec? this doesn't really seem kubernetes-specific.

Reply:
Moved it to be a top-level field of `ClusterSpec` in 2f4a0b5e5